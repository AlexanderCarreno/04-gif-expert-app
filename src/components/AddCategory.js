import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({
  setCategories
}) => {

const [inputValue, setInputValue] = useState('');

const handleInputChange = (e)=> {
    //console.log(e.target.value);
    setInputValue( e.target.value );

    //console.log('handleInputChanges llamado');
}

const handleSubtim = (e) => {
    e.preventDefault();
    //console.log('handleSubmit', inputValue);

    if( inputValue.trim().length > 2 ){
      setCategories( cats => [inputValue, ...cats]);
      setInputValue('');
      //console.log('Submit hecho');
    }
}

  return (
    <>
    <h1>{inputValue}</h1>
    <form onSubmit={ handleSubtim }>
      <p>{ inputValue }</p>
        <input
            type="text"
            value= { inputValue }
            onChange={ handleInputChange }
        />
    </form>
    </>
  )
}


AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}