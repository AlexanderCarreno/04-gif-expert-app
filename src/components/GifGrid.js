import React, { useEffect, useState } from 'react'
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( {category} ) => {
  

  //const [images, setImages] = useState([]);
  //useEffect( () => {
  //  getGifs( category )
  //    .then( imgs => {
  //      //setImages( imgs );
  //      setImages
  //    });
  //}, [ category ]);



  const { data:images, loading } = useFetchGifs( category );


/*
  return (
    <>
      <h3>GifGrid</h3>
      <ol>
        {
          images.map( img => ( 
            <li key={ img.id }>{ img.title }</li>
          ))
        }
      </ol>
    </>
  )
  */

  return (
    <>
      <h3 className='animate__animated animate__fadeIn animate__delay-1s'>{ category }</h3>
      { /*loading ? <p>Loading</p> : null*/ }
      { loading && <p className='animate__animated animate__flash'>Loading</p> }

      <div className='card-grid'>
        {
          images.map( img => ( 
            <GifGridItem 
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    
      {/*

      <div className='card-grid'>
        <ol>
          {
            images.map( img => ( 
              <GifGridItem 
                key={ img.id }
                { ...img }
              />
            ))
          }
        </ol>
      </div>*/
      }
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}