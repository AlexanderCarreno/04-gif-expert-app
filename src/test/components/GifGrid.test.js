import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('debe mostrar el componente <GifGrid /> correctamente', () => {
    const category = 'My Hero Academia';
    
    test('debe mostrar <GifGrid /> corractamente', () => { 

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        
        const wrapper = shallow(<GifGrid category={ category }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
        
        const gifs = [
            {
                id:'ABC',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            },
            {
                id:'12345',
                url: 'https://localhost/cualquier/cosa.jpg',
                title: 'Cualquier cosa'
            },
    ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={ category }/> );
        //expect(wrapper).toMatchSnapshot();
        expect(   wrapper.find('p').exists()  ).toBe(false);
        expect(  wrapper.find('GifGridItem').length  ).toBe( gifs.length );
    });
    
});