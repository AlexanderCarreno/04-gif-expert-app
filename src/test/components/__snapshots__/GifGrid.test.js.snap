// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe mostrar el componente <GifGrid /> correctamente debe mostrar <GifGrid /> corractamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    My Hero Academia
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
